components/secret-mgt/org.wso2.carbon.identity.secret.mgt.core/src/main/java/org/wso2/carbon/identity/secret/mgt/core/SecretManagerImpl.java

```diff
@@ -0,0 +1,311 @@
+import org.wso2.carbon.identity.secret.mgt.core.dao.SecretDAO;
+import org.wso2.carbon.identity.secret.mgt.core.exception.SecretManagementClientException;
+import org.wso2.carbon.identity.secret.mgt.core.exception.SecretManagementException;
+import org.wso2.carbon.identity.secret.mgt.core.exception.SecretManagementServerException;
+import org.wso2.carbon.identity.secret.mgt.core.internal.SecretManagerComponentDataHolder;
+import org.wso2.carbon.identity.secret.mgt.core.model.Secret;
+import org.wso2.carbon.identity.secret.mgt.core.model.Secrets;
+
+import static org.wso2.carbon.identity.secret.mgt.core.constant.SecretConstants.DB_TABLE_NAME;
+import static org.wso2.carbon.identity.secret.mgt.core.constant.SecretConstants.ErrorMessages.ERROR_CODE_GET_DAO;
+import static org.wso2.carbon.identity.secret.mgt.core.constant.SecretConstants.ErrorMessages.ERROR_CODE_INVALID_SECRET_ID;
+import static org.wso2.carbon.identity.secret.mgt.core.constant.SecretConstants.ErrorMessages.ERROR_CODE_SECRET_ADD_REQUEST_INVALID;
+import static org.wso2.carbon.identity.secret.mgt.core.constant.SecretConstants.ErrorMessages.ERROR_CODE_SECRET_ALREADY_EXISTS;
+import static org.wso2.carbon.identity.secret.mgt.core.constant.SecretConstants.ErrorMessages.ERROR_CODE_SECRET_DELETE_REQUEST_REQUIRED;
+import static org.wso2.carbon.identity.secret.mgt.core.constant.SecretConstants.ErrorMessages.ERROR_CODE_SECRET_DOES_NOT_EXISTS;
+import static org.wso2.carbon.identity.secret.mgt.core.constant.SecretConstants.ErrorMessages.ERROR_CODE_SECRET_GET_REQUEST_INVALID;
+import static org.wso2.carbon.identity.secret.mgt.core.constant.SecretConstants.ErrorMessages.ERROR_CODE_SECRET_MANAGER_NOT_ENABLED;
+import static org.wso2.carbon.identity.secret.mgt.core.constant.SecretConstants.ErrorMessages.ERROR_CODE_SECRET_REPLACE_REQUEST_INVALID;
+import static org.wso2.carbon.identity.secret.mgt.core.util.SecretUtils.generateUniqueID;
+import static org.wso2.carbon.identity.secret.mgt.core.util.SecretUtils.handleClientException;
+import static org.wso2.carbon.identity.secret.mgt.core.util.SecretUtils.handleServerException;
+
+/**
+ * Secret Manager service implementation.
+ */
+public class SecretManagerImpl implements SecretManager {
+
+    private static final Log log = LogFactory.getLog(SecretManagerImpl.class);
+    private List<SecretDAO> secretDAOS;
+
+    public SecretManagerImpl() {
+
+        this.secretDAOS = SecretManagerComponentDataHolder.getInstance().getSecretDAOS();
+    }
+
+    @Override
+    public Secret addSecret(Secret secret) throws SecretManagementException {
+
+        validateSecretManagerEnabled();
+        validateSecretCreateRequest(secret);
+        String secretId = generateUniqueID();
+        if (log.isDebugEnabled()) {
+            log.debug("Secret id generated: " + secretId);
+        }
+        secret.setSecretId(secretId);
+        this.getSecretDAO().addSecret(secret);
+        log.info("Secret: " + secret.getSecretName() + " added successfully");
```

Comment:
Let's change this to debug.

We can combine the previous debug and have it as a single debug log

Reply:
let's add debug logs for other methods as well.

Reply:
resolved